packages/react-components/lib/doors/door-panel.tsx

```diff
@@ -14,16 +14,26 @@ import React from 'react';
 import * as RmfModels from 'rmf-models';
 import { DoorTable } from './door-table';
 import { DoorData, doorModeToString, doorTypeToString } from './utils';
+import { FixedSizeGrid, GridChildComponentProps } from 'react-window';
+import AutoSizer from 'react-virtualized-auto-sizer';
 
 export interface DoorPanelProps {
   doors: DoorData[];
   doorStates: Record<string, RmfModels.DoorState>;
   onDoorControlClick?(event: React.MouseEvent, door: RmfModels.Door, mode: number): void;
 }
 
-export interface DoorInfoProps {
+export interface DoorGridDataProps extends DoorPanelProps {
+  columnCount: number;
+}
+
+export interface DoorGridRendererProps extends GridChildComponentProps {
+  data: DoorGridDataProps;
+}
+
+export interface DoorcellProps {
   door: DoorData;
-  doorMode?: number;
+  doorState?: RmfModels.DoorState;
```

Comment:
Using an object here breaks memo as each update is a different object, revert back to `doorMode` or add custom `React.memo` logic.